blockchain/emulator.go

```diff
@@ -0,0 +1,270 @@
+
+	args, err := parseArguments(arguments)
+	if err != nil {
+		return nil, nil, err
+	}
+	tx.Arguments = args
+
+	return e.sendTransaction(tx, authorizers)
+}
+
+func (e *emulator) executeScript(script string, arguments []string) (*types.ScriptResult, error) {
+	args, err := parseArguments(arguments)
+	if err != nil {
+		return nil, err
+	}
+
+	return e.blockchain.ExecuteScript([]byte(script), args)
+}
+
+func (e *emulator) createAccount() (*flowsdk.Account, *flowsdk.Transaction, *types.TransactionResult, error) {
+	payer := e.blockchain.ServiceKey().Address
+
+	key := flowsdk.NewAccountKey()
+	key.FromPrivateKey(e.blockchain.ServiceKey().PrivateKey)
+	key.HashAlgo = crypto.SHA3_256
+	key.SetWeight(flowsdk.AccountKeyWeightThreshold)
+
+	tx, err := templates.CreateAccount([]*flowsdk.AccountKey{key}, nil, payer)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	result, tx, err := e.sendTransaction(tx, nil)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	account := &flowsdk.Account{
+		Address: parseEventAddress(result.Events),
+	}
+
+	return account, tx, result, nil
+}
+
+func (e *emulator) getAccount(address flowsdk.Address) (*flowsdk.Account, *emu.AccountStorage, error) {
+	storage, err := e.blockchain.GetAccountStorage(address)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	account, err := e.blockchain.GetAccount(address)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return account, storage, nil
+}
+
+func (e *emulator) deployContract(
+	address flowsdk.Address,
+	script string,
+) (*types.TransactionResult, *flowsdk.Transaction, error) {
+	contractName, err := getSourceContractName(script)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	tx := templates.AddAccountContract(address, templates.Contract{
+		Name:   contractName,
+		Source: script,
+	})
+
+	return e.sendTransaction(tx, nil)
+}
+
+func (e *emulator) sendTransaction(
+	tx *flowsdk.Transaction,
+	authorizers []flowsdk.Address,
+) (*types.TransactionResult, *flowsdk.Transaction, error) {
+	signer, err := e.blockchain.ServiceKey().Signer()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "error getting service signer")
+	}
+
+	for _, auth := range authorizers {
+		tx.AddAuthorizer(auth)
+	}
+	tx.SetPayer(e.blockchain.ServiceKey().Address)
+
+	// we translate addresses from client address space to the emulator space
+	// client uses address starting at 0x01 whereas emulator starts at 0x05
+	r := regexp.MustCompile("0x0+([1-9])+")
+	found := r.FindAllStringSubmatch(string(tx.Script), -1)
+
+	if len(found) > 0 {
+		for _, f := range found {
+			// if found a match for address then convert to number and convert to emulator address space by the address offset
+			addressOffset := 4
+			addressNumber, _ := strconv.Atoi(f[1])
+			original := f[0]
+			replaced := strings.ReplaceAll(original, fmt.Sprintf("%d", addressNumber), fmt.Sprintf("%d", addressNumber+addressOffset))
+			tx.Script = []byte(strings.ReplaceAll(string(tx.Script), original, replaced))
+		}
+	}
```

Comment:
For sure. I haven't yet because I still am not too happy about it. I'm waiting EOW when @janezpodhostnik mentioned he could have an FVM improvement allowing us to specify to not take this extra accounts up.